arduino/libraries/loader.go

```diff
@@ -126,6 +127,11 @@ func makeNewLibrary(libraryDir *paths.Path, location LibraryLocation) (*Library,
 }
 
 func makeLegacyLibrary(path *paths.Path, location LibraryLocation) (*Library, error) {
+	if foundHeader, err := containsHeaderFile(path); err != nil {
+		return nil, err
+	} else if !foundHeader {
+		return nil, errors.Errorf(tr("library not valid"))
```

Comment:
Let's make it a bit more verbose on why it has been rejected.
```suggestion
		return nil, errors.Errorf(tr("invalid library: no header files found"))
```